client/my-sites/sidebar/publish-menu.jsx

```diff
@@ -124,84 +100,96 @@ var PublishMenu = React.createClass( {
 
 		// Hide the sidebar link for multiple site view if it's not in calypso, or
 		// if it opts not to be shown.
+		const isEnabled = config.isEnabled( menuItem.config );
 		if ( ! this.props.isSingle && ( ! isEnabled || ! menuItem.showOnAllMySites ) ) {
 			return null;
 		}
 
-		if ( ! isEnabled && this.props.site.options ) {
+		let link;
+		if ( ! isEnabled && site.options ) {
 			link = this.props.site.options.admin_url + menuItem.wpAdminLink;
 		} else {
 			link = menuItem.link + this.props.siteSuffix;
 		}
 
 		let preload;
-
-		if ( menuItem.name === 'post' ) {
-			icon = 'posts';
+		if ( includes( [ 'post', 'page' ], menuItem.name ) ) {
 			preload = 'posts-pages';
-		} else if ( menuItem.name === 'page' ) {
-			icon = 'pages';
-			preload = 'posts-pages';
-		} else if ( menuItem.name === 'jetpack-portfolio' ) {
-			icon = 'folder';
-		} else if ( menuItem.name === 'jetpack-testimonial' ) {
-			icon = 'quote';
 		} else {
-			icon = 'custom-post-type';
+			preload = 'posts-custom';
+		}
+
+		let icon;
+		switch ( menuItem.name ) {
+			case 'post': icon = 'posts'; break;
+			case 'page': icon = 'pages'; break;
+			case 'jetpack-portfolio': icon = 'folder'; break;
+			case 'jetpack-testimonial': icon = 'quote'; break;
+			default: icon = 'custom-post-type';
 		}
 
+		const className = this.props.itemLinkClass(
+			menuItem.paths ? menuItem.paths : menuItem.link,
+			menuItem.className
+		);
+
 		return (
-			<SidebarItem key={ menuItem.name }
+			<SidebarItem
+				key={ menuItem.name }
 				label={ menuItem.label }
 				className={ className }
 				link={ link }
 				buttonLink={ menuItem.buttonLink }
-				onNavigate={ this.props.onNavigate }
+				onNavigate={ this.onNavigate.bind( this, menuItem.name ) }
 				icon={ icon }
 				preloadSectionName={ preload }
 			/>
 		);
 	},
 
-	render: function() {
-		var menuItems = this.getDefaultMenuItems( this.props.site ),
-			customMenuItems,
-			customPostTypes;
-
-		customPostTypes = this.state.postTypes.filter( function( type ) {
-			return ! some( menuItems, { name: type.name } );
-		} );
-
-		customMenuItems = customPostTypes.map( function( postType ) {
+	getCustomMenuItems() {
+		const customPostTypes = omit( this.props.postTypes, [ 'post', 'page' ] );
+		return map( customPostTypes, function( postType ) {
 			return {
 				name: postType.name,
-				label: postType.labels ? postType.labels.menu_name : postType.label,
+				label: get( postType.labels, 'menu_name', postType.label ),
 				className: postType.name,
+				config: 'manage/custom-post-types',
```

Comment:
So this is still a dummy config correct?  Do you foresee us maybe having config flags on a per CPT basis?